### PR TITLE
Conf hooks

### PR DESCRIPTION
**Pull request by marodru**

### CODE REVIEW

File: applications/crossbar/src/modules/cb_webhooks.erl [hunk 1]
```diff
@@ -497,6 +500,16 @@ check_modifier_values(CustomValue, ModifierKey, ModifierValue, <<"object">>, Con
             cb_context:add_validation_error(ModifierKey, <<"enum">>, Msg, Context);
         _ ->
             Context
+    end;
+check_modifier_values(CustomValue, ?CONFERENCE_ID_MODIFIER, _ModifierValue, <<"string">>, Context) ->
+    case check_conference_id(CustomValue, Context) of
+        'false' ->
+            Msg = kz_json:from_list([{<<"message">>, <<"specified conference id is not valid">>}
+                ,{<<"conference_id">>, CustomValue}
```
Comment (kalda341) [inline applications/crossbar/src/modules/cb_webhooks.erl]: Running the `make fmt` command the way it is run in `circle.yml` will fix formatting issues like this

File: applications/webhooks/src/webhooks_util.erl [hunk 2]
```diff
@@ -141,6 +143,50 @@ match_subaccount_webhooks(HookEvent, AccountId) ->
...
+    MatchSpec = [{
+        #webhook{hook_event='$1',_='_'}
+        ,[{'==','$1', {'const', HookEvent}}]
+        ,['$_']
+    }],
+    Hooks = ets:select(table_id(), MatchSpec),
+    match_conference_webhooks_by_name(Hooks, ConferenceId, CommandName).
+match_conference_webhooks_by_name(Hooks, ConferenceId, CommandName) ->
+    FilteredHooks = lists:foldl(
+        fun(#webhook{custom_data=Data}=Webhook, Acc) ->
```
Comment (kalda341) [inline applications/webhooks/src/webhooks_util.erl]: I'd probably use a named function here

File: applications/webhooks/src/webhooks_util.erl [hunk 3]
```diff
@@ -141,6 +143,50 @@ match_subaccount_webhooks(HookEvent, AccountId) ->
...
+            case Data of
+                undefined-> Acc;
+                {DataList}->
```
Comment (kalda341) [inline applications/webhooks/src/webhooks_util.erl]: Why not keep it as JSON and use the kz_json functions?

File: applications/webhooks/src/webhooks_util.erl [hunk 4]
```diff
@@ -141,6 +143,50 @@ match_subaccount_webhooks(HookEvent, AccountId) ->
...
+                    end
+            end
+        end
+        , []
+        , Hooks),
+    match_conference_webhooks_by_id(FilteredHooks, ConferenceId).
+match_conference_webhooks_by_id(Hooks, ConferenceId) ->
+    lists:foldl(
+        fun(#webhook{custom_data=Data}=Webhook, Acc) ->
+            {DataList} = Data,
```
Comment (kalda341) [inline applications/webhooks/src/webhooks_util.erl]: Here too

**Comment by kalda341** [comment]
@marodru I've made a few suggestions, but otherwise it looks good :)

File: applications/crossbar/src/modules/cb_webhooks.erl (hunk 1 above)
Comment (serg-i-o) [reply applications/crossbar/src/modules/cb_webhooks.erl]: Ok. I've just fixed only my project files by fix.sh, cause using make fix make changes in whole module crossbar

File: applications/webhooks/src/webhooks_util.erl (hunk 2 above)
Comment (serg-i-o) [reply applications/webhooks/src/webhooks_util.erl]: I completely refactored match_conference_* functions in webhook_utils and combine two searches in one function chain - match_conf_foldl(...) -> match_conf_by_key(...)

File: applications/webhooks/src/webhooks_util.erl (hunk 3 above)
Comment (serg-i-o) [reply applications/webhooks/src/webhooks_util.erl]: I think that using json will be more expensive than using stdlibs lists:keyfind. If we will using json we must convert Hooks list to json, then make search (that will be same as lists:keyfind), and then make back convert json result to the hook. 
With a large number of conferences, events can occur very often and each search will be costly for the kapps instance.

**Comment by serg-i-o** [comment]
I've just make fixes. Please check code again.

**Pull request merged**